src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,252 @@
+            rowData: { rows: data, row: row },
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: RowData,
+        column: Column
+    ) => {
+        if (column?.parse) {
+            row[column.name] = column.parse(event.target.value, row, column);
+        } else if (event.target.type === "number") {
+            row[column.name] = Number(event.target.value);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        raiseEvent("custom", {
+            eventType: "cellChange",
+            rowData: { rows: data, row: row, column: column },
+        });
+        setData([...data]);
+    };
+
+    return (
+        <Box maxHeight={maxHeight} maxWidth={maxWidth} sx={{ overflow: "auto" }}>
+            <Table
+                size={size}
+                stickyHeader={stickyHeader}
+                sx={{
+                    // Hack required to override .gcx-forms.defaults
+                    ["& span"]: {
+                        display: "inline-flex !important",
+                    },
+                }}
+            >
+                <TableHead>
+                    <TableRow>
+                        {includeFooter && footerLabel && <TableCell />}
+                        {cols.map((col) => (
+                            <TableCell align={col.align} key={col.name}>
+                                {col.alias ?? col.name}
+                            </TableCell>
+                        ))}
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {data.map((row, index) => {
+                        return (
+                            <TableRow
+                                hover
+                                key={index}
+                                onMouseEnter={(event) => handleMouseEnter(event, row)}
+                                onMouseLeave={(event) => handleMouseLeave(event, row)}
+                            >
+                                {includeFooter && footerLabel && <TableCell />}
+
+                                {cols.map((col, index) => {
+                                    const value = row[col.name];
+                                    const formattedValue = col.format
+                                        ? col.format(value, row)
+                                        : value;
+                                    return (
+                                        <TableCell align={col.align} key={index}>
+                                            {col.editable ? (
+                                                <Input
+                                                    fullWidth={col.type === "text" ? true : false}
+                                                    type={col.type}
+                                                    value={formattedValue}
```

Comment:
I think we need to re-think the automatically editable input. If someone's formatter is doing something like calculating a percentage with the % symbol, it is going to require a really nasty combination of `parse` and `format` functions that match one another perfectly.

We need to talk to PXD.

Reply:
Maybe it's best to remove format and parse on inputs and just let the input do it's job.  In fact we can do away with parse altogether. The only issue is really with decimals and presenting them in a way that's consistent with what a user expects (i.e. currency).  On an uneditable cell, we can apply format only since we're always working with the data, not the input.   Perhaps we expose the step property as a optional input so that we can set the number of decimal places?